src/resolver.rs

```diff
@@ -1,31 +1,64 @@
-use std::{collections::HashMap, sync::Arc};
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
 
 use async_recursion::async_recursion;
-use miette::{ensure, Context, Diagnostic};
+use miette::{bail, ensure, Context, Diagnostic};
 use semver::VersionReq;
 use thiserror::Error;
 
 use crate::{
     cache::{Cache, Entry},
     credentials::Credentials,
     lock::{FileRequirement, Lockfile},
-    manifest::{Dependency, Manifest},
-    package::{Package, PackageName},
+    manifest::{
+        Dependency, DependencyManifest, LocalDependencyManifest, Manifest,
+        RemoteDependencyManifest, MANIFEST_FILE,
+    },
+    package::{Package, PackageName, PackageStore},
     registry::{Artifactory, RegistryUri},
 };
 
 /// Represents a dependency contextualized by the current dependency graph
-pub struct ResolvedDependency {
-    /// The materialized package as downloaded from the registry
-    pub package: Package,
-    /// The registry the package was downloaded from
-    pub registry: RegistryUri,
-    /// The repository in the registry where the package can be found
-    pub repository: String,
-    /// Packages that requested this dependency (and what versions they accept)
-    pub dependants: Vec<Dependant>,
-    /// Transitive dependencies
-    pub depends_on: Vec<PackageName>,
+pub enum ResolvedDependency {
+    /// A resolved dependency that is located on a remote registry
+    Remote {
+        /// The materialized package as downloaded from the registry
+        package: Package,
+        /// The registry the package was downloaded from
+        registry: RegistryUri,
+        /// The repository in the registry where the package can be found
+        repository: String,
+        /// Packages that requested this dependency (and what versions they accept)
+        dependants: Vec<Dependant>,
+        /// Transitive dependencies
+        depends_on: Vec<PackageName>,
+    },
+    /// A resolved depnedency that is located on the filesystem
+    Local {
+        /// The materialized package that was created from the buffrs package at the given path
+        package: Package,
+        /// Location of the requested package
+        path: PathBuf,
+        /// Packages that requested this dependency (and what versions they accept)
+        dependants: Vec<Dependant>,
```

Comment:
What about just a `Vec<String>`? I believe the version range is always going to be `*`

Reply:
Although this is correct im in favor of using types to communicate what is inside of the `String`